Source/RxSwift/toggleBool.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  toggleBool.swift
+//  RxSwiftExt-iOS
+//
+//  Created by Vincent Pradeilles on 20/02/2018.
+//  Copyright © 2018 RxSwiftCommunity. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+extension ObservableType {
+    
+    /**
+     Returns an observable sequence of booleans, starting with `initial` and toggling its value
+     at each `.next` event
+     
+     - parameter initial: the boolean value to initialize the sequence
+     - returns: An observable sequence of booleans that starts with `initial` and toogles its value at each `.next` event
+     */
+    public func toggleBool(initial: Bool) -> Observable<Bool> {
+        return scan(initial) { previous, _ in return !previous }.startWith(initial)
```

Comment:
Implementation question for both yourself and @fpillet 
I think the startWith shouldn't be part of this operator necessarily, since scan itself has a seed value. 

It creates a scenario that by providing 4 input values, ill get 5 output emissions 

e.g. 1 2 3 4 (seed = false)
emit: false true false true false
Where the first two would happen together.

I'd expect it to be just `false, true, false, true` while the first one emits only upon the emitting of the source sequence. What do you guys think? Does either make more sense than the other?

Reply:
To me it made sense to have the initial value, because I was working with UI, so I expected the resulting Observable to have an initial value, like a Behavior.

Reply:
But would you expect an operator to emit something without any values going through it? It seems confusing to me. It means an empty sequence will still emit the initial value.

Reply:
If the function were to return an `BehaviorSubject` instead of an `Observable`, would this behavior then be deemed more acceptable?

Reply:
> But would you expect an operator to emit something without any values going through it? It seems confusing to me. It means an empty sequence will still emit the initial value.

Fully agree

Reply:
Yup, it's been a long time since we discussed this one. I don't think it needs an initial value.